sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs

```diff
@@ -220,7 +221,7 @@ public enum SasProtocol
     }
     public partial class TableSasBuilder
     {
-        public TableSasBuilder(string tableName) { }
+        public TableSasBuilder(string tableName, Azure.Data.Tables.Sas.TableSasPermissions permissions, System.DateTimeOffset expiresOn) { }
```

Comment:
Aren't the permissions optional in some cases depending on your ACL?  Might be cleaner to just leave this .ctor as default and rely on the TableClient.GetSasBuilder method for the convenience.

Reply:
I believe for [service level sas tokens](https://docs.microsoft.com/en-us/rest/api/storageservices/create-service-sas#construct-a-service-sas), it is required.  My thinking was to put all the required elements into the ctor to help devs avoid a situation where we let them build an invalid token or return an error when we validate. Thoughts?

Reply:
Ah, in that case you should have one overload for each set of required parameters that are valid together.  Something like `TableSasBuilder(permissions, expires)` and `TableSasBuiler(name, expires)`.

Since ACLs are less common and not using `name` and `permissions` together might be kind of confusing, it might be worth making it optional so you had `TableSasBuilder(expires, permissions)` and `TableSasBuiler(name, expires, permissions = null)`.   I don't feel very strongly about this since you've still got the `SetPermissions` methods.

Reply:
We also probably want a `TableServiceClient.GetSasBuilder` for constructing service level SAS tokens.

Reply:
> We also probably want a `TableServiceClient.GetSasBuilder` for constructing service level SAS tokens.

Based on the [discussion thread from the previous PR](https://github.com/Azure/azure-sdk-for-net/pull/11766#discussion_r419668582) the current thought is to direct service level scenarios to Storage.Common. Do you feel that is still ok?

Reply:
> Ah, in that case you should have one overload for each set of required parameters that are valid together. Something like `TableSasBuilder(permissions, expires)` and `TableSasBuiler(name, expires)`.
> 
> Since ACLs are less common and not using `name` and `permissions` together might be kind of confusing, it might be worth making it optional so you had `TableSasBuilder(expires, permissions)` and `TableSasBuiler(name, expires, permissions = null)`. I don't feel very strongly about this since you've still got the `SetPermissions` methods.

As I understand it, all three are always required.

Reply:
Account level != service level.  Account level is for all storage resources (blobs, queues, files, tables) in your account.  Service level is for any table in your account.

Reply:
Ah, thanks for the clarification!

Reply:
Resolving per our offline conversation